analysis/estimate/notebooks/run-estimate.Rmd

```diff
@@ -0,0 +1,289 @@
+---
+title: "Run ESTIMATE on bulk and pseudobulk samples"
+author: "Stephanie J. Spielman"
+date: "`r Sys.Date()`"
+output:
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    code_folding: hide
+---
+
+The purpose of this notebook is to run ESTIMATE on bulk (counts and TPM) and pseudobulk samples and compare results to one another.
+
+## Setup
+
+
+```{r setup}
+renv::load()
+
+library(ggplot2)
+library(tidyestimate)
+theme_set(theme_bw())
+```
+
+### Paths
+
+```{r paths}
+data_dir <- here::here("analysis", "estimate", "data")
+sample_metadata_file <- here::here("s3_files", "scpca-sample-metadata.tsv")
+library_metadata_file <- here::here("s3_files", "scpca-library-metadata.tsv")
+expression_files <- list.files(
+  data_dir, 
+  pattern = "_expression\\.rds$"
+) |>
+  purrr::set_names(
+    \(x) {stringr::str_split_i(x, pattern = "_", i = 1)}
+  )
+
+output_file <- here::here("analysis", "estimate", "results", "estimate_results.tsv")
+```
+
+### Read input data
+
+```{r}
+sample_metadata_df <- readr::read_tsv(sample_metadata_file, show_col_types = FALSE)
+library_metadata_df <- readr::read_tsv(library_metadata_file, show_col_types = FALSE)
+expression_data <- expression_files |>
+  purrr::map(
+    \(filename) {
+      readr::read_rds(file.path(data_dir, filename))
+})
+```
+
+## Run ESTIMATE
+
+We will now run ESTIMATE on all three types of counts for samples in each project:
+
+* bulk counts
+* bulk TPM
+* pseudobulk counts
+
+```{r, message = FALSE}
+estimate_df <- expression_data |>
+  # map over projects
+  purrr::map(
+    \(project_df_list) {
+      # map over expression types
+      
+      purrr::map(project_df_list, \(expr_df) {
+        ## Run ESTIMATE
+        expr_df |> 
+          dplyr::rename(hgnc_symbol = gene_symbol) |>
+          tidyestimate::filter_common_genes(
+            id = "hgnc_symbol", 
+            tidy = TRUE, # first column contains gene symbol
+            tell_missing = FALSE, 
+            find_alias = TRUE
+          ) |> 
+          tidyestimate::estimate_score(is_affymetrix = FALSE)
+      }) |> 
+        purrr::list_rbind(names_to = "expression_type")
+  }) |> 
+  purrr::list_rbind(names_to = "project_id") |>
+  dplyr::rename(sample_id = sample) |>
+  # this is just the sum of scores, we won't be working with it
+  dplyr::select(-estimate) |> 
+  # make scores long too
+  tidyr::pivot_longer(
+    cols = c(stromal, immune), 
+    names_to = "score_type", 
+    values_to = "score"
+  )
+```
+
+
+## Visualization
+
+
+Let's begin by looking at the overall values:
```

Comment:
I might switch the order of this notebook a little bit. 
- first look at the relationship between bulk TPM and counts 
- then pick one and show this sina plot for just the counts and pseudobulk followed by the comparison of counts to pseudobulk.